crypto/x509/x509_test.cc

```diff
@@ -4923,26 +4923,38 @@ TEST(X509Test, SignatureVerification) {
                    X509_V_FLAG_PARTIAL_CHAIN));
 
   // Bad keys in the root and intermediate are rejected.
-  EXPECT_EQ(X509_V_ERR_UNABLE_TO_DECODE_ISSUER_PUBLIC_KEY,
+  EXPECT_EQ(X509_V_ERR_UNSPECIFIED,
             Verify(leaf.valid.get(), {root.bad_key.get()},
                    {intermediate.valid.get()}, {}));
-  EXPECT_EQ(X509_V_ERR_UNABLE_TO_DECODE_ISSUER_PUBLIC_KEY,
+  EXPECT_EQ(X509_V_ERR_UNSPECIFIED,
             Verify(leaf.valid.get(), {root.bad_key_type.get()},
                    {intermediate.valid.get()}, {}));
-  EXPECT_EQ(X509_V_ERR_UNABLE_TO_DECODE_ISSUER_PUBLIC_KEY,
+  EXPECT_EQ(X509_V_ERR_UNSPECIFIED,
             Verify(leaf.valid.get(), {root.valid.get()},
                    {intermediate.bad_key.get()}, {}));
-  EXPECT_EQ(X509_V_ERR_UNABLE_TO_DECODE_ISSUER_PUBLIC_KEY,
+  EXPECT_EQ(X509_V_ERR_UNSPECIFIED,
```

Comment:
From a first glance, the error messages seem to become more unclear due to the changes. Is that intentional for more OpenSSL compatibility?

Reply:
This would actually be more inline with the OpenSSL behavior which returns unspecified at this point if the return code < 0.